### PR TITLE
Stethoscopes and penlights - An attempt on usefulness (Also, flashlight flashing fixed)

### PR DESCRIPTION
**Pull request by LC4492**
# About the pull request

The main purpose of this PR is to add functionality to both the stethoscope and the penlight, with the fixing of the "flash eyes with flashlight" mechanism being only a conterminous addon. 

Special thanks to Drathek for assisting with coding this, without his help I wouldn't go anywhere.

# Explain why it's good for the game

Adds more detail to the game and allows the player to have more options regarding the diagnosis of their patients. Even though this PR is not meant to replace any type of current diagnosis methods, it is thought to allow the possibility of these items being used if the player wish, or if the situation they are in requires. It is also thought to actually give some purpose to both items, instead of only being decorative (basically).

About the "flash eyes with flashlight" mechanism, it just have been repaired because of its relation with the functionality of the penlight itself. (Remember, only the penlight can be used to do diagnosis, generic flashlights will only allow to temporarily blind the person you are interacting with, and will not give you any medical information).

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

image

image

image

image

image

</details>

# Changelog
:cl:
add: Adds penlights and stethoscopes to nurses and corpsmen, they can use them to reliably check if someone have specific types of organ damage and how much, without the need of other apparatus. Adds a new organ.status called "LITTLE_BRUISED", used by the stethoscope and penlight to not give false positives when diagnosticating someone (saying that they are healthy, when they actually have 9 heart damage).
fix: Fixed an outdated check that made the entire "flash eyes with flashlight" mechanic don't work at all. Also updated it to check some other things for ease of logic.
spellcheck: Fixed typos on the abandoned "flash eyes with flashlight" code.
code: Added new functionalities for both the stethoscope and penlight: The stethoscopes can now be used to check the condition of both heart and lungs individually by aiming the chest, letting you be able to check if the organs are LITTLE_BRUISED (Have a damage equal or above 1), BRUISED (Have a damage equal or above 10), BROKEN (have a damage equal or above 30) or "HEALTHY" (any damage below 1). And the penlight, that can be used the same way to check the condition of brain and eyes by aiming the eyes of the person you are interacting with. Also with the fixing mentioned above, you can now flash people's eyes again! Everytime you check the condition of someone's eyes, you will also flash them. Replaced one letter vars in the flashlight code and on the stethoscope section of the ties code.
imageadd: Adds new "in_hand" icons for penlights and stethoscopes, both will now appear on your hands when used. Penlights have both "off" and "on" icons, fancy stuff!
/:cl:

### CODE REVIEW

**Comment by hislittlecuzingames** [comment]
Maybe I missed it but... can pen be held in ear?
Can stethoscope be held on face slot? (Around neck)

May be storage creep, but maybe cool.

File: code/game/machinery/vending/vendor_types/crew/medical.dm [hunk 1]
```diff
@@ -245,6 +245,9 @@ GLOBAL_LIST_INIT(cm_vending_clothing_researcher, list(
 		/obj/item/tool/surgery/synthgraft,
 		/obj/item/storage/syringe_case,
 		/obj/item/storage/surgical_case/regular,
+		/obj/item/clothing/accessory/stethoscope,
+		/obj/item/device/flashlight/pen/,
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/crew/medical.dm:249]:
```suggestion
		/obj/item/device/flashlight/pen,
```

File: code/game/objects/items/devices/flashlight.dm [hunk 2]
```diff
@@ -100,37 +100,40 @@
...
-/obj/item/device/flashlight/attack(mob/living/M as mob, mob/living/user as mob)
+/obj/item/device/flashlight/attack(mob/living/carbon/human/M as mob, mob/living/user as mob)
 	add_fingerprint(user)
 	if(on && user.zone_selected == "eyes")
 
 		if((user.getBrainLoss() >= 60) && prob(50)) //too dumb to use flashlight properly
 			return ..() //just hit them in the head
 
-		if((!ishuman(user) || SSticker) && SSticker.mode.name != "monkey") //don't have dexterity
+		if (!(istype(user, /mob/living/carbon/human) || SSticker) && SSticker.mode.name != "monkey") //don't have dexterity
```
Comment (Zonespace27) [inline code/game/objects/items/devices/flashlight.dm:110]: why did you replace this

File: code/game/machinery/vending/vendor_types/squad_prep/squad_medic.dm [hunk 3]
```diff
@@ -236,4 +236,6 @@ GLOBAL_LIST_INIT(cm_vending_clothing_medic, list(
 		/obj/item/storage/surgical_case/regular,
 		/obj/item/reagent_container/blood/OMinus,
 		/obj/item/reagent_container/blood/OMinus,
+		/obj/item/device/flashlight/pen/,
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_medic.dm:239]:
```suggestion
		/obj/item/device/flashlight/pen,
```

File: code/game/objects/items/devices/flashlight.dm [hunk 4]
```diff
@@ -147,14 +150,76 @@
 
 /obj/item/device/flashlight/pen
 	name = "penlight"
-	desc = "A pen-sized light, used by medical staff."
+	desc = "A pen-sized light, used by medical staff to check the condition of eyes, brain and cosncience of patients."
```
Comment (Zonespace27) [inline code/game/objects/items/devices/flashlight.dm:153]: conscience is both misspelled and the wrong word to use here
also oxford comma

File: code/game/objects/items/devices/flashlight.dm [hunk 5]
```diff
@@ -147,14 +150,76 @@
...
+		if(istype(M, /mob/living/carbon/human)) //robots and aliens are unaffected
+			var/datum/internal_organ/eyes/eyes = M.internal_organs_by_name["eyes"]
+			var/datum/internal_organ/brain/brain = M.internal_organs_by_name["brain"]
+			var/reaction = "try to watch closely, but you see no difference in their eyes' reaction"
+			if(isnull(M.internal_organs_by_name))
+				reaction = "discover that indeed he has nothing to check"
+				return // they have no organs somehow
+			if(M == user) //they're using it on themselves
+				M.flash_eyes()
+				M.visible_message(SPAN_NOTICE("[M] directs [src] to \his eyes."), \
```
Comment (Zonespace27) [inline code/game/objects/items/devices/flashlight.dm:177]: we have a gender macro that works better than \his, see line 188

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/flashlight.dm (hunk 2 above)
Comment (LC4492) [reply code/game/objects/items/devices/flashlight.dm:110]: That's the exact code line that wasn't allowing the flashing to work, so I simply replaced it with a more modern counterpart, and now works without any problems at all

Comment (LC4492) [reply code/game/objects/items/devices/flashlight.dm:110]: The older line is possibly outdated

File: code/game/objects/items/devices/flashlight.dm (hunk 4 above)
Comment (LC4492) [reply code/game/objects/items/devices/flashlight.dm:153]: Yes, will replace it with a more decent line

File: code/game/objects/items/devices/flashlight.dm (hunk 5 above)
Comment (LC4492) [reply code/game/objects/items/devices/flashlight.dm:177]: I know, that is still part of the older flashlight flashing code, I didn't replace it on the philosophy that you shouldn't be replacing stuff that is working, but I can easily replace it with a "[M.p_their()]", no problems at all in that

**Comment by blackdragonTOW** [comment]
Hello! Please remember to attach screenshots of your altered sprites in the PR description!!

**Comment by LC4492** [comment]
> Hello! Please remember to attach screenshots of your altered sprites in the PR description!!

I did add the images, but it seems they are not showing up in github properly

**Comment by LC4492** [comment]
> > Hello! Please remember to attach screenshots of your altered sprites in the PR description!!
> 
> I did add the images, but it seems they are not showing up in github properly

All showing properly now

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/game/objects/items/devices/flashlight.dm (hunk 5 above)
Comment (Zonespace27) [reply code/game/objects/items/devices/flashlight.dm]: Yes, do replace it

File: code/game/objects/items/devices/flashlight.dm (hunk 2 above)
Comment (Zonespace27) [reply code/game/objects/items/devices/flashlight.dm:110]: The ishuman macro will 100% work

**Review by nauticall — APPROVED** [review]
sprites

File: code/game/objects/items/devices/flashlight.dm (hunk 2 above)
Comment (LC4492) [reply code/game/objects/items/devices/flashlight.dm:110]: > The ishuman macro will 100% work

I have tested it myself, and that's the cause of the malfunction

**Review by kiVts — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/game/objects/items/devices/flashlight.dm [hunk 6]
```diff
@@ -147,14 +150,85 @@
...
+	throw_speed = SPEED_VERY_FAST
+	throw_range = 15
+	matter = list("metal" = 10,"glass" = 5)
 	raillight_compatible = 0
 
+/obj/item/device/flashlight/pen/attack(mob/living/carbon/human/being as mob, mob/living/user as mob)
+	add_fingerprint(user)
+	if(user.a_intent == INTENT_HELP)
+		if(on && user.zone_selected == "eyes")
+			if(ishuman_strict(being)) //robots and aliens are unaffected
```
Comment (kiVts) [inline code/game/objects/items/devices/flashlight.dm:169]: Still not an early return

Comment (kiVts) [reply code/game/objects/items/devices/flashlight.dm:169]: 😈

Comment (kiVts) [inline code/game/objects/items/devices/flashlight.dm:169]:
```suggestion
			if(!ishuman_strict(being)) //robots and aliens are unaffected
```

File: code/game/objects/items/devices/flashlight.dm [hunk 7]
```diff
@@ -73,68 +73,71 @@
...
 
-		user.visible_message(SPAN_NOTICE("[user] directs [src] to [M]'s eyes."), \
-							SPAN_NOTICE("You direct [src] to [M]'s eyes."))
+		user.visible_message(SPAN_NOTICE("[user] directs [src] to [being]'s eyes."), \
+							SPAN_NOTICE("You direct [src] to [being]'s eyes."))
 
-		if(istype(M, /mob/living/carbon/human)) //robots and aliens are unaffected
-			if(M.stat == DEAD || M.sdisabilities & DISABILITY_BLIND) //mob is dead or fully blind
-				to_chat(user, SPAN_NOTICE("[M] pupils does not react to the light!"))
+		if(istype(being, /mob/living/carbon/human)) //robots and aliens are unaffected
```
Comment (kiVts) [inline code/game/objects/items/devices/flashlight.dm:128]:
```suggestion
		if(ishuman_strict(being)) //robots and aliens are unaffected
```

File: code/game/objects/items/devices/flashlight.dm [hunk 8]
```diff
@@ -147,14 +150,85 @@
...
+							if(ORGAN_BRUISED)
+								being.flash_eyes()
+								reaction = "observe that [being.p_their()] eyes are <font color='orange'>unrealiably reacting to the light</font>, with [being.p_their()] pupils <font color='orange'>reacting very sluggishly and with noticeable delays</font>, it is probable that [being.p_their()] vision is <font color='orange'>remarkably impaired</font>"
+							if(ORGAN_BROKEN)
+								reaction = "notice that [being.p_their()] eyes are <font color='red'>not reacting to the light</font>, and the pupils of both eyes are <font color='red'>not constricting with the light</font> shine at all, [being.p_they()] is probably <font color='red'>blind</font>"
+							else
+								being.flash_eyes()
+								reaction = "perceive that [being.p_their()] eyes and pupils are <font color='green'>normally reacting to the light</font>, [being.p_they()] is probably<font color='green'> seeing without problems</font>"
+					if(brain)
```
Comment (kiVts) [inline code/game/objects/items/devices/flashlight.dm:199]: final thing, check if the reaction exists here and if it does just add ", You also". There is a lot of copypaste code here for essentially same thing but adds ". You also" on top of that, reaction is never zero, so I dont see the purpose of the checks below

File: code/modules/clothing/under/ties.dm [hunk 9]
```diff
@@ -102,42 +102,68 @@
...
+											sound = "hear <font color='orange'>deviant heart beating patterns</font>, result of probable <font color='orange'>heart damage</font>"
+										if(ORGAN_BROKEN)
+											sound = "hear <font color='red'>irregular and additional heart beating patterns</font>, probably caused by impaired blood pumping, [being.p_their()] heart is certainly <font color='red'>failing</font>"
+										else
+											sound = "hear <font color='green'>normal heart beating patterns</font>, [being.p_their()] heart is surely <font color='green'>healthy</font>"
+								var/datum/internal_organ/lungs/lungs = being.internal_organs_by_name["lungs"]
+								if(lungs)
+									switch(lungs.organ_status)
+										if(ORGAN_LITTLE_BRUISED)
+											if(sound)
```
Comment (kiVts) [inline code/modules/clothing/under/ties.dm:139]: same here, lots of copypaste that could be easily avoided

**Review by kiVts — CHANGES_REQUESTED** [review]
one final thing I promise!!!

File: code/game/objects/items/devices/flashlight.dm (hunk 8 above)
Comment (kiVts) [reply code/game/objects/items/devices/flashlight.dm:199]: so add `if (reaction) reaction += ". You also"` right after if(brain), and clean up the text below, I believe in you, mark as ready when done

**Review by kiVts — APPROVED** [review]
looks good

File: code/game/objects/items/devices/flashlight.dm (hunk 2 above)
Comment (kiVts) [reply code/game/objects/items/devices/flashlight.dm]: resolved

File: code/game/objects/items/devices/flashlight.dm (hunk 5 above)
Comment (kiVts) [reply code/game/objects/items/devices/flashlight.dm]: resolved

**Comment by Toastina** [comment]
Late to the party but. can you code so that penlight works inside the helmet. it's effect is similar to rail light

**Pull request merged**